navigation/eskf/include/eskf/eskf_ros.hpp

```diff
@@ -0,0 +1,69 @@
+#ifndef ESKF_ROS_HPP
+#define ESKF_ROS_HPP
+
+#include <chrono>
+#include <geometry_msgs/msg/pose_with_covariance_stamped.hpp>
+#include <geometry_msgs/msg/twist_with_covariance_stamped.hpp>
+#include <nav_msgs/msg/odometry.hpp>
+#include <rclcpp/rclcpp.hpp>
+#include <sensor_msgs/msg/imu.hpp>
+#include <std_msgs/msg/bool.hpp>
+#include <std_msgs/msg/float64_multi_array.hpp>
+#include <std_msgs/msg/string.hpp>
+#include <variant>
+#include "eskf/eskf.hpp"
+#include "eskf/typedefs.hpp"
+#include "typedefs.hpp"
+
+class ESKFNode : public rclcpp::Node {
```

Comment:
Might want to let nom/error state be private member variables of this class with their own getters, would avoid **needing** to return it everywhere and making the user define this on the outside. This is "less" monadic than the current impl, but I think nicer overall.